core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -240,6 +240,16 @@ private[spark] object Utils extends Logging {
     // scalastyle:on classforname
   }
 
+  /**
+   * Run a segment of code using a different context class loader in the current thread
+   */
+  def withContextClassLoader(ctxClassLoader: ClassLoader)(fn: => Unit): Unit = {
```

Comment:
Better to make this method generic (i.e. make the return a parameterized type `T`).